app/src/main/java/com/example/places/MapFragment.kt

```diff
@@ -40,23 +39,24 @@ class MapFragment : Fragment(R.layout.map_fragment), OnMapReadyCallback {
     override fun onMapReady(p0: GoogleMap) {
         p0.let {
             googleMap = it
-            viewModel.placesLiveData.observe(this, { places ->
+            viewModel.placesLiveData.observe(this) { places ->
                 markerPlaces(places)
-            })
-
+            }
+            viewModel.selectedPlace.observe(this) { place ->
+                var location = LatLng(place.latitude.toDouble(),place.longitude.toDouble())
+                zoomPlace(location)
+            }
         }
     }
-
     fun markerPlaces(placesList: List<Place>) {
-        for (places in placesList) {
-            if (places.latitude != null && places.longitude != null) {
-                val location = LatLng(places.latitude.toDouble(), places.longitude.toDouble())
-                googleMap.addMarker(MarkerOptions().position(location))
-                googleMap.animateCamera(CameraUpdateFactory.newLatLngZoom(location, 10f))
-            }
-
+        for (place in placesList) {
+            val location = LatLng(place.latitude.toDouble(),place.longitude.toDouble())
+            googleMap.addMarker(MarkerOptions().position(location))
+            zoomPlace(location)
```

Comment:
You're zooming in to every place in this for loop. This is unnecessary work. Just zoom in at one of them at random, or at the first, or the last.